File: api-tester/src/main/java/com/revenuecat/apitester/java/EntitlementInfosAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.EntitlementInfo;
import com.revenuecat.purchases.EntitlementInfos;
import com.revenuecat.purchases.VerificationResult;

import java.util.Map;

@SuppressWarnings({"unused", "deprecation"})
final class EntitlementInfosAPI {
    static void check(final EntitlementInfos infos) {
        final Map<String, EntitlementInfo> active = infos.getActive();
        final Map<String, EntitlementInfo> all = infos.getAll();
        final EntitlementInfo i = infos.get("");
        final VerificationResult verification = infos.getVerification();
    }

    static void checkConstructor(
            Map<String, EntitlementInfo> all,
            VerificationResult verificationResult
    ) {
        final EntitlementInfos entitlementInfos = new EntitlementInfos(
                all,
                verificationResult
        );
        final EntitlementInfos entitlementInfos2 = new EntitlementInfos(all);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/EntitlementInfosAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.EntitlementInfo
import com.revenuecat.purchases.EntitlementInfos
import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI
import com.revenuecat.purchases.VerificationResult

@Suppress("unused", "UNUSED_VARIABLE", "DEPRECATION")
private class EntitlementInfosAPI {
    fun check(infos: EntitlementInfos) {
        val active: Map<String, EntitlementInfo> = infos.active
        val all: Map<String, EntitlementInfo> = infos.all
        val i: EntitlementInfo? = infos[""]
        val verification: VerificationResult = infos.verification
    }

    fun checkConstructor(
        all: Map<String, EntitlementInfo>,
        verificationResult: VerificationResult,
    ) {
        val entitlementInfos = EntitlementInfos(
            all = all,
            verification = verificationResult,
        )
        val entitlementInfos2 = EntitlementInfos(all = all)
    }
}
